doc/api/process.md

```diff
@@ -2305,6 +2305,222 @@ import { loadEnvFile } from 'node:process';
 loadEnvFile();
 ```
 
+### Environment variables file parser specification
+
+This section describes how the environment variables file parser reads a file
+and sets up the environment variables in Node.js.
+
+1. **Basic parsing**:
+
+   * The parser reads line by line, splitting each line into a key and a value
+     at the first `=` sign.
+   * Lines without an `=` are ignored.
+
+   ```bash
+      BASIC=basic
+      WITHOUT_EQUAL
+   ```
+
+   ```cjs
+   const assert = require('node:assert');
+   const process = require('node:process');
+   assert.strictEqual(process.env.BASIC, 'basic');
+   assert.strictEqual(process.env.WITHOUT_EQUAL, undefined);
+   ```
+
+   ```mjs
+   import assert from 'node:assert';
+   import process from 'node:process';
+   assert.strictEqual(process.env.BASIC, 'basic');
+   assert.strictEqual(process.env.WITHOUT_EQUAL, undefined);
+   ```
+
+2. **Comments**:
+
+   * Lines starting with `#` are treated as comments and ignored.
+   * Inline comments (starting with `#` after a value) are ignored, and do not
+     affect parsing of the value.
+
+   ```bash
+       # COMMENTS=work
+       INLINE_COMMENTS=inline comments # work
```

Comment:
Is the space after `comments` part of the value of `INLINE_COMMENTS`?

Reply:
@GeoffreyBooth No, we trim spaces in both the key and the value